src/dxgi/dxgi_monitor.cpp

```diff
@@ -142,6 +145,26 @@ namespace dxvk {
                                  / pMode->RefreshRate.Denominator;
     }
     
+    /* Compare new mode with current and change if necessary */
+    if (EnumDisplaySettingsW(monInfo.szDevice, ENUM_CURRENT_SETTINGS, &currentMode)) {
+      if (currentMode.dmPelsWidth == devMode.dmPelsWidth &&
+        currentMode.dmPelsHeight == devMode.dmPelsHeight &&
+        currentMode.dmBitsPerPel == devMode.dmBitsPerPel &&
+        (currentMode.dmDisplayFrequency == devMode.dmDisplayFrequency ||
+        !(devMode.dmFields & DM_DISPLAYFREQUENCY)) &&
+        (currentMode.dmDisplayFlags == devMode.dmDisplayFlags ||
+        !(devMode.dmFields & DM_DISPLAYFLAGS)))
```

Comment:
`(devMode.dmFields & DM_DISPLAYFLAGS)` is always `false`, so this check should probably be removed.